src/main.rs

```diff
@@ -21,6 +30,99 @@ struct Args {
 
     #[clap(short = 'o', long = "output-format", default_value = "pretty")]
     output_format: LogOutputFormat,
+
+    /// TODO: remove once config is ready
+    #[clap(long = "signers-xonly")]
+    signers_xonly: String,
+}
+
+async fn fetch_and_create_deposits(
+    context: &Context,
+    deposit_monitor: &DepositMonitor,
+    chain_tip: &BlockRef,
+) -> Result<(), Error> {
+    let emily_config = context.emily_config();
+
+    let deposits = deposit_monitor.get_pending_deposits(chain_tip)?;
+
+    tracing::debug!(count = deposits.len(), "fetched pending deposits");
+    if deposits.is_empty() {
+        return Ok(());
+    }
+
+    for deposit in deposits {
+        // TODO: emily will nop for duplicates, but we shouldn't send them
+        if let Err(error) = deposit_api::create_deposit(emily_config, deposit.clone()).await {
+            tracing::warn!(
+                %error,
+                txid = %deposit.bitcoin_txid,
+                vout = %deposit.bitcoin_tx_output_index,
+                "cannot create deposit in emily"
+            );
+        } else {
+            tracing::info!(
+                txid = %deposit.bitcoin_txid,
+                vout = %deposit.bitcoin_tx_output_index,
+                "created deposit in emily"
+            );
+        }
+    }
+
+    Ok(())
+}
+
+async fn runloop(context: Context, deposit_monitor: &DepositMonitor, polling_interval: Duration) {
+    let bitcoin_client = context.bitcoin_client();
+    let mut last_chain_tip = None;
+
+    loop {
+        if last_chain_tip.is_some() {
+            tokio::time::sleep(polling_interval).await;
+        }
+
+        let chain_tip = match bitcoin_client.get_chain_tip() {
+            Ok(chain_tip) => chain_tip,
+            Err(error) => {
+                tracing::warn!(
+                    %error,
+                    "error getting the chain tip"
+                );
+                continue;
+            }
+        };
+
+        if last_chain_tip
+            .as_ref()
+            .is_some_and(|last| last == &chain_tip)
```

Comment:
nit: can we break this out into two steps, a variable assignment and then the if statement.